raft/raft.cpp

```diff
@@ -60,7 +62,14 @@ raft::raft(
     this->state_files_exist() ? this->load_state()
                                : this->create_state_files(log_path, peers);
 
-    this->raft_log = std::make_shared<bzn::raft_log>(log_path);
+    this->raft_log = std::make_shared<bzn::raft_log>(log_path, maximum_raft_storage);
+
+    // KEP-112 Bail if the raft storage exceeds the max storage
+    if(this->raft_log->maximum_storage_exceeded())
+    {
+        LOG(error) << MSG_ERROR_MAXIMUM_STORAGE_EXCEEDED;
+        throw std::runtime_error(MSG_ERROR_MAXIMUM_STORAGE_EXCEEDED);
```

Comment:
should we use just one way to exit?

Reply:
I want to leave this one as a throw simply because I was able to unit test the functionality easily...

I'd like to push back on this one, but, it's easy enough to replace the throw with raise(SIGINT)...